packages/block-library/src/navigation/edit/index.js

```diff
@@ -655,6 +667,16 @@ function Navigation( {
 							/* translators: %s: The name of a menu. */
 							actionLabel={ __( "Switch to '%s'" ) }
 						/>
+						<BlockEditorProvider
+							value={ navigationInnerBlocks }
+							onChange={ onChange }
+							onInput={ onInput }
+						>
+							<ListView
+								blocks={ navigationInnerBlocks }
+								isExpanded={ true }
+							/>
+						</BlockEditorProvider>
```

Comment:
In this case the inner blocks are uncontrolled, so I don't think `BlockEditorProvider` is needed (there isn't an entity to edit with `onChange` and `onInput`).